libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -43,11 +48,33 @@ type Enabled struct {
 
 type PluginConfig []map[string]common.Config
 
+func getSystemKubeConfig() string {
+	envKubeConfig := os.Getenv("KUBECONFIG")
+	if _, err := os.Stat(envKubeConfig); !os.IsNotExist(err) {
+		return envKubeConfig
+	}
+	homeKubeConfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	if _, err := os.Stat(homeKubeConfig); !os.IsNotExist(err) {
+		return homeKubeConfig
+	}
+	return ""
+}
+
+func inCluster() bool {
+	_, err := rest.InClusterConfig()
+	if err == nil {
+		return true
+	}
+	return false
+}
+
 func defaultKubernetesAnnotatorConfig() kubeAnnotatorConfig {
 	return kubeAnnotatorConfig{
+		KubeConfig:      getSystemKubeConfig(),
```

Comment:
I think that in other places we are assuming that if no kubeconfig is specified, then in cluster configuration is wanted. We should keep this behaviour also here.

Reply:
I think it falls under the same doubts of https://github.com/elastic/beats/pull/13473#discussion_r323198447